### PR TITLE
get-kube-binaries: use GCE token to fetch artifacts from GCS

### PR DESCRIPTION
**Pull request by javier-b-perez**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
GCE should use credentials to fetch objects from GCS

**Special notes for your reviewer**:
Prow is using get-kube.sh script to create clusters.
This change should allow prow to use private GCS objects.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/get-kube-binaries.sh [hunk 1]
```diff
@@ -119,13 +119,24 @@ function sha1sum_file() {
   fi
 }
 
+# Get default service account credentials of the VM.
+function get-credentials {
+  curl "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token" -H "Metadata-Flavor: Google" -s | python -c \
```
Comment (mikedanese) [inline cluster/get-kube-binaries.sh]: What OS environment does prow run inside of? Is gcloud installed?

**Comment by mikedanese** [comment]
Can you guarantee that this script never runs with set -o xtrace? If it is, you will dump access tokens to the log. Alternatively, you could opportunistically swap curl for gsutil if it's available in the environment running get-kube-binaries.

File: cluster/get-kube-binaries.sh (hunk 1 above)
Comment (krzyzacy) [reply cluster/get-kube-binaries.sh]: prow runs a pod with whatever image user defines, we will ensure to use an image with gcloud for our use case here - also I think optionally use gsutil might be a better idea

Comment (javier-b-perez) [reply cluster/get-kube-binaries.sh]: we don't really need gcloud, it is just a local endpoint in the VM

File: cluster/get-kube-binaries.sh [hunk 2]
```diff
@@ -143,13 +143,34 @@ function sha1sum_file() {
...
+}
+
+function valid-storage-scope {
+  curl "${GCE_METADATA_INTERNAL}/service-accounts/default/scopes" -H "Metadata-Flavor: Google" -s | grep -q "auth/devstorage"
+}
+
 function download_tarball() {
   local -r download_path="$1"
   local -r file="$2"
+  local trace_on="off"
```
Comment (javier-b-perez) [inline cluster/get-kube-binaries.sh]: @mikedanese do you think this will cover it?

**Comment by krzyzacy** [comment]
(friendly poke ^^)

File: cluster/get-kube-binaries.sh (hunk 2 above)
Comment (mikedanese) [reply cluster/get-kube-binaries.sh]: What does this do?

File: cluster/get-kube-binaries.sh [hunk 3]
```diff
@@ -143,13 +143,34 @@ function sha1sum_file() {
...
+    trace_on="on"
+    set +x
+  fi
   url="${DOWNLOAD_URL_PREFIX}/${file}"
   mkdir -p "${download_path}"
   if [[ $(which curl) ]]; then
-    curl -fL --retry 3 --keepalive-time 2 "${url}" -o "${download_path}/${file}"
+    # if the url belongs to GCS API we should use oauth2_token in the headers
+    local curl_headers=""
+    if [[ "${KUBERNETES_PROVIDER:-gce}" == "gce" ]] && [[ "$url" =~ ^https://storage.googleapis.com.* ]] && valid-storage-scope ; then
```
Comment (krzyzacy) [inline cluster/get-kube-binaries.sh]: ooops, also gke?

**Comment by krzyzacy** [comment]
^^
/hold

File: cluster/get-kube-binaries.sh (hunk 3 above)
Comment (javier-b-perez) [reply cluster/get-kube-binaries.sh]: done

**Pull request merged**